-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47321/43647

ERROR: Build errors found during clang-tidy run.
```
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02876/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/alloc_traits.h:518:4: error: no matching function for call to 'construct_at' [clang-diagnostic-error]
  518 |           std::construct_at(__p, std::forward<_Args>(__args)...);
      |           ^~~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02876/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_tree.h:597:23: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<std::_Rb_tree_node<std::pair<const unsigned int, std::vector<unsigned int>>>>>::construct<std::pair<const unsigned int, std::vector<unsigned int>>, unsigned int, const unsigned int &, int>' requested here
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/RecoLuminosity/LumiProducer/plugins/TRGScalers2DB.cc
+++ b/RecoLuminosity/LumiProducer/plugins/TRGScalers2DB.cc
@@ -621,35 +621,25 @@ namespace lumi {
       TRGScalers2DB::TriggerDeadCountResult::iterator dEnd = deadtimeresult.end();
       unsigned int dcnt = 0;
       for (dIt = dBeg; dIt != dEnd; ++dIt) {
-        try {
-          deadfracresult.at(dcnt);
-        } catch (std::out_of_range& er) {
+        if (dcnt >= deadfracresult.size()) {
           std::cout << "[WARNING] filling FAKE deadfrac=0.0 at LS " << dcnt << std::endl;
-          deadfracresult[dcnt] = 0.0;
+          deadfracresult.push_back(0.0);
         }
-        try {
-          algocount.at(dcnt);
-        } catch (std::out_of_range& er) {
-          std::vector<unsigned int> tmpzero(lumi::N_TRGALGOBIT, 0);
+        if (dcnt >= algocount.size()) {
           std::cout << "[WARNING] filling FAKE algocount at LS " << dcnt << std::endl;
-          algocount[dcnt] = tmpzero;
+          algocount.emplace_back(lumi::N_TRGALGOBIT, 0);
         }
-        try {
-          techcount.at(dcnt);
-        } catch (std::out_of_range& er) {
-          std::vector<unsigned int> tmpzero(lumi::N_TRGTECHBIT, 0);
+        if (dcnt >= techcount.size()) {
           std::cout << "[WARNING] filling FAKE techcount at LS " << dcnt << std::endl;
-          techcount[dcnt] = tmpzero;
+          techcount.emplace_back(lumi::N_TRGTECHBIT, 0);
         }
         if (algoprescale.find(dcnt + 1) == algoprescale.end()) {
-          std::vector<unsigned int> tmpzero(lumi::N_TRGALGOBIT, 1);
           std::cout << "[WARNING] filling FAKE 1 algoprescale at LS " << dcnt + 1 << std::endl;
-          algoprescale[dcnt + 1] = tmpzero;
+          algoprescale[dcnt + 1] = std::vector<unsigned int>(lumi::N_TRGALGOBIT, 1);
         }
         if (techprescale.find(dcnt + 1) == techprescale.end()) {
-          std::vector<unsigned int> tmpzero(lumi::N_TRGTECHBIT, 1);
           std::cout << "[WARNING] filling FAKE 1 techprescale at LS " << dcnt + 1 << std::endl;
-          techprescale[dcnt + 1] = tmpzero;
+          techprescale[dcnt + 1] = std::vector<unsigned int>(lumi::N_TRGTECHBIT, 1);
         }
         ++dcnt;
       }

